packages/plugin-connector/src/plugin-discovery/get-root-modules.js

```diff
@@ -0,0 +1,40 @@
+import path from 'path';
+import fs from 'fs/promises';
+
+/**
+ * @param {string} p
+ * @returns {Promise<boolean>}
+ */
+const hasNodeModules = async (p) => {
+	const directoryItems = await fs.readdir(p);
+	return directoryItems.includes('node_modules');
+};
+
+/**
+ * Attempts to find the highest, project-scoped node_modules filepath.
+ * Note that this behavior _should_ remain the same regardless of where this is run from
+ * Locally installed ("linked") packages and regularly installed packages should return the same value.
+ * @param {string | undefined} [startingPoint]
+ * @returns {Promise<string>}
+ */
+export const getRootModules = async (startingPoint) => {
+	// Either use the entry file or a specific startingPoint
+	const entryFile = startingPoint ?? process.cwd();
```

Comment:
I think this should be okay, but how trustworthy is process.cwd?

Reply:
If we run this as a top level workspace command, cwd is not trustworthy. This shouldn't be a _major_ issue, but some thought would be worthwhile.